src/test/HostActivationTests/GivenThatICareAboutNativeHostApis.cs

```diff
@@ -52,29 +52,91 @@ public void Muxer_activation_of_Publish_Output_Portable_DLL_hostfxr_get_native_s
                 .HaveStdOutContaining("hostfxr_get_native_search_directories buffer:[" + dotnet.GreatestVersionSharedFxPath);
         }
 
+        [Fact]
+        public void Breadcrumb_thread_finishes_when_app_closes_normally()
+        {
+            var fixture = sharedTestState.PreviouslyPublishedAndRestoredPortableAppProjectFixture.Copy();
+            var dotnet = fixture.BuiltDotnet;
+            var appDll = fixture.TestProject.AppDll;
+
+            dotnet.Exec(appDll)
+                .EnvironmentVariable("CORE_BREADCRUMBS", sharedTestState.BreadcrumbLocation)
+                .EnvironmentVariable("COREHOST_TRACE", "1")
+                .CaptureStdOut()
+                .CaptureStdErr()
+                .Execute()
+                .Should()
+                .Pass()
+                .And
+                .HaveStdOutContaining("Hello World")
+                .And
+                .HaveStdErrContaining("Waiting for breadcrumb thread to exit...");
+        }
+
+        [Fact]
+        public void Breadcrumb_thread_does_not_finish_when_app_has_unhandled_exception()
+        {
+            var fixture = sharedTestState.PreviouslyPublishedAndRestoredPortableAppWithExceptionProjectFixture.Copy();
+            var dotnet = fixture.BuiltDotnet;
+            var appDll = fixture.TestProject.AppDll;
+
+            dotnet.Exec(appDll)
+                .EnvironmentVariable("CORE_BREADCRUMBS", sharedTestState.BreadcrumbLocation)
+                .EnvironmentVariable("COREHOST_TRACE", "1")
+                .CaptureStdOut()
+                .CaptureStdErr()
+                .Execute(fExpectedToFail: true)
+                .Should()
+                .Fail()
+                .And
+                .HaveStdErrContaining("Unhandled Exception: System.Exception: Goodbye World")
+                .And
+                // The breadcrumb thread does not wait since destructors are not called when an exception is thrown.
+                // However, destructors will be called when the caller (such as a custom host) is compiled with SEH Exceptions (/EHa) and has a try\catch.
+                // Todo: add a native host test app so we can verify this behavior.
+                .NotHaveStdErrContaining("Waiting for breadcrumb thread to exit...");
```

Comment:
Do you want to also assert that there ARE breadcrumb messages getting written?  Just not this one...

Reply:
The test is to ensure the destructor is called in the happy path, and not called in the exception case (for the normal host that doesn't have SEH exception catching support), and checking for that string is sufficient.

Also I think adding a probe to see if a physical file would be too intrusive. The breadcrumb files are only written if they do not exist, so that would mean we'd have to delete existing files global files (or change the date on them to an earlier date, since a touch is done if they already exist).